src/erlcloud_workspaces.erl

```diff
@@ -0,0 +1,446 @@
+    out(Return, fun(Json, UOpts) -> decode_record(describe_workspaces_record(), Json, UOpts) end, 
+        WorkspacesOpts).
+
+%%%------------------------------------------------------------------------------
+%% DescribeTags
+%%%------------------------------------------------------------------------------
+-type describe_tags_opt() :: {resource_id, string_param()} |
+                             out_opt().
+-type describe_tags_opts() :: [describe_tags_opt()].
+
+-spec describe_tags_opts() -> opt_table().
+describe_tags_opts() ->
+    [
+        {resource_id, <<"ResourceId">>, fun to_binary/1}
+    ].
+
+-spec describe_tags(Opts :: describe_tags_opts()) -> workspaces_return([#workspaces_tag{}]).
+describe_tags(Opts) ->
+    describe_tags(Opts, default_config()).
+
+%%%------------------------------------------------------------------------------
+%% @doc 
+%% Workspaces API
+%% [https://docs.aws.amazon.com/workspaces/latest/api/API_DescribeTags.html]
+%%
+%% ===Example===
+%%
+%% Describe tags for workspace id "ws-c8wvb67py"
+%%
+%% `
+%% {ok, Tags} = erlcloud_workspaces:describe_tags([{resource_id, "ws-c8wvb67py"}, {out, json}])
+%% '
+%% @end
+%%%------------------------------------------------------------------------------
+-spec describe_tags(Opts :: describe_tags_opts(), Config :: aws_config()) -> workspaces_return([#workspaces_tag{}]).
+describe_tags(Opts, #aws_config{} = Config) ->
+    {AwsOpts, WorkspacesOpts} = opts(describe_tags_opts(), Opts),
+    Return = workspaces_request(
+                Config,
+                "DescribeTags",
+                AwsOpts),
+    out(Return, fun(Json, UOpts) ->
+            TagsList = proplists:get_value(<<"TagList">>, Json),
+            decode_tags_list(TagsList, UOpts)
+        end, 
+        WorkspacesOpts).
+
+%%%------------------------------------------------------------------------------
+%%% Internal Functions
+%%%------------------------------------------------------------------------------
+workspaces_request(Config, Operation, Body) ->
+    case erlcloud_aws:update_config(Config) of
+        {ok, Config1} ->
+            workspaces_request_no_update(Config1, Operation, Body);
+        {error, Reason} ->
+            {error, Reason}
+    end.
+
+workspaces_request_no_update(Config, Operation, Body) ->
+    Payload = case Body of
+               [] -> <<"{}">>;
+               _ -> jsx:encode(lists:flatten(Body))
+           end,
+    Headers = headers(Config, Operation, Payload),
+    Request = #aws_request{service = workspaces,
+                           uri = uri(Config),
+                           method = post,
+                           request_headers = Headers,
+                           request_body = Payload},
+    case erlcloud_aws:request_to_return(erlcloud_retry:request(Config, Request, fun workspaces_result_fun/1)) of
+        {ok, {_RespHeaders, <<>>}} -> {ok, []};
+        {ok, {_RespHeaders, RespBody}} -> {ok, jsx:decode(RespBody, [{return_maps, false}])};
```

Comment:
Any reason not to make this `true`? What's the point of adding new APIs with the legacy format?

Reply:
I just followed the [best practices](https://github.com/erlcloud/erlcloud#best-practices). It doesn't tell anything about maps. 

> Public interfaces should normally expose proplists over records. All new modules are preferred to have both.

Also this implementation was borrowed from `erlcloud_ecs.erl`

Reply:
@motobob should we change this or just keep digging the 'proplist grave' for 'consistency'?

Reply:
@nalundgaard I've added `{out, map}` option. Please let me know what you think?

Reply:
LGTM.